### PR TITLE
Let `use_package()` decrement a package dependency.

### LINKED ISSUE #1957: use_package() `min_version` will not decrement
When `min_version` is provided and it is a smaller semver than the one currently in the DESCRIPTION file, it will not be decremented to the lower version. 

```r
usethis::use_package("dplyr", min_version = "1.0.0")
usethis::use_package("dplyr", min_version = "0.8.0")
```

### PR DESCRIPTION
**Pull request by jplecavalier**
Fixes #1957.

### CODE REVIEW

File: R/helpers.R [hunk 1]
```diff
@@ -62,19 +62,19 @@ use_dependency <- function(package, type, min_version = NULL) {
...
-      ui_bullets(c(
-        "v" = "Increasing {.pkg {package}} version to {.val {version}} in
-               DESCRIPTION."
-      ))
-      desc$set_dep(package, type, version = version)
-      desc$write()
-      changed <- TRUE
-    }
+    direction <- ifelse(version_spec(version) > version_spec(existing_version),
+                        "Increasing", "Decreasing")
```
Comment (jennybc) [inline R/helpers.R:68]: Can you change this to an `if () ... else`? I generally reserve `ifelse()` for vectorized situations.

File: R/helpers.R [hunk 2]
```diff
@@ -62,19 +62,19 @@ use_dependency <- function(package, type, min_version = NULL) {
...
+
+    ui_bullets(c(
+      "v" = "{.drn {direction}} {.pkg {package}} version to {.val {version}} in
```
Comment (jennybc) [inline R/helpers.R:71]: I'm not familiar with the `.drn` inline markup. Can you help me understand that?

**Review by jennybc — COMMENTED** [review]
Looking good! I had a couple comments/questions.

File: R/helpers.R (hunk 1 above)
Comment (jplecavalier) [reply R/helpers.R:68]: Sure!

File: R/helpers.R (hunk 2 above)
Comment (jplecavalier) [reply R/helpers.R:71]: I just tried to replicate the same logic as in `{.pkg {package}}` and`{.val {version}}`. I tried to remove `.drn` and then `{direction}` would not resolve with the string in the object `direction`. Any hint?

Comment (jennybc) [reply R/helpers.R]: Something like `{.pkg {package}}` does interpolation of `package` through its surrounding brackets. Then `.pkg` and the second layer of brackets applies some inline styling.

For pure interpolation (which is the goal with `direction`), we just need one pair of brackets.

You can learn more about this in the cli package: https://cli.r-lib.org/reference/inline-markup.html.

Comment (jplecavalier) [reply R/helpers.R]: Thanks for pointing this out, and thanks again for your time and patience today!

**Pull request merged**